src/index.ts

```diff
@@ -192,14 +198,55 @@ module.exports = class ServerlessPlugin {
       if (config.subscribeToExecutionLogs) {
         await addExecutionLogGroupsAndSubscriptions(this.serverless.service, aws, datadogForwarderArn);
       }
+
+      if (config.subscribeToStepFunctionLogs) {
+        const functionArn = datadogForwarderArn;
+        const resources = this.serverless.service.provider.compiledCloudFormationTemplate?.Resources;
+        const stepFunctions = Object.values((this.serverless.service as any).stepFunctions.stateMachines);
+        if (stepFunctions.length === 0) {
+          this.serverless.cli.log(
+            "Warning: subscribeToStepFunctionLogs is set to true but no step functions were found.",
+          );
+        } else {
+          this.serverless.cli.log("Subscribing step function log groups to Datadog Forwarder");
+          for (const stepFunction of stepFunctions as any[]) {
+            if (!stepFunction.hasOwnProperty("loggingConfig")) {
+              this.serverless.cli.log(
+                `Warning: No logging config found for step function ${stepFunction.name}. Creating log group and logging to it with level ALL.`,
+              );
+              await addStepFunctionLogGroup(aws, resources, stepFunction);
+            } else {
+              this.serverless.cli.log(`Found logging config for step function ${stepFunction.name}`);
+              const loggingConfig = stepFunction.loggingConfig;
+
+              if (loggingConfig.level !== "ALL") {
+                loggingConfig.level = "ALL";
+                this.serverless.cli.log(
+                  `Warning: Setting log level to ALL for step function ${stepFunction.name} so traces can be generated.`,
+                );
+              }
+              if (loggingConfig.includeExecutionData !== true) {
+                loggingConfig.includeExecutionData = true;
+                this.serverless.cli.log(
+                  `Warning: Setting includeExecutionData to true for step function ${stepFunction.name} so traces can be generated.`,
+                );
+              }
+            }
+            // subscribe step function log group to datadog forwarder regardless of how the log group was created
+            await addStepFunctionLogGroupSubscription(resources, stepFunction, functionArn);
+          }
+        }
+      }
       for (const error of errors) {
         this.serverless.cli.log(error);
       }
     }
 
     if (datadogForwarderArn && config.addExtension) {
       this.serverless.cli.log(
-        "Warning: Datadog Lambda Extension and forwarder are both enabled. Only APIGateway log groups will be subscribed to the forwarder.",
+        `Warning: Datadog Lambda Extension and forwarder are both enabled. Only APIGateway ${
+          config.subscribeToStepFunctionLogs ? "and Step Function " : ""
+        }log groups will be subscribed to the forwarder.`,
```

Comment:
I thought it might be confusing for the customer if the following situation arose:

* Forwarder and Extension both enabled
* Step function log subscription enabled
* Log message for `...Only APIGateway log groups will be subscribed to the forwarder.` appears during deployment

Based on the log message, it seems like step function log groups were not subscribed to the forwarder even though they really were. The updated log message would read `...Only APIGateway and Step Function log groups will be subscribed to the forwarder.`

Reply:
Got it. Makes sense.